### PR TITLE
Update CoC according to suggestions from current CoC committee

### PR DESCRIPTION
**Pull request by jennaswa**
The current CoC committee (@jennaswa, @micaela-matta, and @richardjgowers) met on September 1, 2023 to discuss proposed updates to the MDAnalysis CoC and CoC procedures. This PR reflects the proposed changes agreed upon by the committee and requires approval from all core developers.

Changes made in this Pull Request:

-Remove option to file a report by e-mail
-Edit language throughout to clarify that CoC Committee is responsible for receiving, investigating, and enforcing reports
-Add reporting procedures for events (and acknowledge NumFOCUS for this text)

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4289.org.readthedocs.build/en/4289/

### CODE REVIEW

**Comment by jennaswa** [comment]
In addition, we propose to set up a conduct@mdanalysis.org account, which would be used to change ownership of the reporting form. There are also a few proposed changes to the questions on the reporting form itself to assist with collecting details relevant to an incident. See our [google doc on CoCC & Procedures](https://docs.google.com/document/d/1N0Pktq5xXXyxVk3yt6Er79nrCmgKm3Kh2IeGJksEJ9A/edit?usp=sharing) for notes on all proposed changes.

**Review by tylerjereddy — APPROVED** [review]
LGTM, +1

**Review by richardjgowers — APPROVED** [review]

**Review by RMeli — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -130,11 +128,22 @@ You may also follow [NumFOCUS’s Code of Conduct violation reporting][NF-conduc
...
 ## Enforcement
 
-When a report is sent to us we will reply as soon as possible to confirm receipt;
-we strive to answer in less than 24 hours. We will review the incident and
-determine, to the best of our ability
+All reports of breaches of the code of conduct will be investigated and handled by the **MDAnalysis Code of Conduct Committee**. The current MDAnalysis Code of Conduct Committee consists of:
+- Jenna Swarthout Goddard (@jennaswa)
+- Micaela Matta (@micaela-matta)
+- Richard Gowers (@richardjgowers)
+In the event of any conflict of interest, the affected committee member will immediately notify the rest of the committee and recuse themselves if necessary.
```
Comment (orbeckst) [inline CODE_OF_CONDUCT.md]: Remove the "if necessary" as it unnecessarily qualifies the initial strong statement. I believe that *if there's a potential conflict of interest* (real or perceived doesn't matter) *then they are just excusing themselves as a matter of policy*.
```suggestion
In the event of any conflict of interest, the affected committee member will immediately notify the rest of the committee and recuse themselves.
```

**Review by orbeckst — APPROVED** [review]
I have one comment where I believe the text needs changing.

But overall this is great and I don't want to hold this up. Thanks for your work, CoC Committee!

**Review by IAlibay — APPROVED** [review]
lgtm - apologies for the delay, I thought I had already approved for some reason

**Review by lilyminium — APPROVED** [review]
LGTM!

**Pull request merged**